test/container_manager/canary_utils.go

```diff
@@ -0,0 +1,109 @@
+package container_manager
+
+import (
+	"flag"
+	"github.com/Shopify/sarama"
+	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/strimzi/strimzi-canary/internal/config"
+	"github.com/strimzi/strimzi-canary/internal/security"
+	"github.com/strimzi/strimzi-canary/internal/servers"
+	"github.com/strimzi/strimzi-canary/internal/services"
+	"github.com/strimzi/strimzi-canary/internal/workers"
+	"log"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+	"time"
+)
+
+var (
+	clientCreationFailed = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name:      "client_creation_error_total",
+		Namespace: "strimzi_canary",
+		Help:      "Total number of errors while creating Sarama client",
+	}, nil)
+)
+
+// TODO goes against DRY principle, possible refactor of cmd/main to put more of code into none main module
+func createCanary() {
+	canaryConfig := config.NewCanaryConfig()
+
+	// Always log to stderr by default
+	if err := flag.Set("logtostderr", "true"); err != nil {
+		glog.Errorf("Error on setting logtostderr to true")
+	}
+	flag.Set("v", strconv.Itoa(canaryConfig.VerbosityLogLevel))
+	flag.Parse()
```

Comment:
Please evaluate if moving flag parsing to a `TestMain(*testing.M)` helps.
The benefit is that it happens at package level, and so you avoid calling
`flag.Parse()` different tests within the same package.

Reply:
both this and next one recommendation are sort of resolved, as I refactored this part of code (replace by running application as different process, which was another suggestion by Paolo)